File: components/MaxWidth/go/component.go
package MaxWidth

import (
	"fmt"
	"html/template"
)

type Instance struct {
	size int32
}

func New() Instance {
	return Instance{}
}
func Defined(size int32) Instance {
	return Instance{size: size}
}

func (i Instance) Size() int32         { return i.size }
func (i *Instance) SetSize(size int32) { i.size = size }

func (i Instance) Html() template.HTML {
	return template.HTML(fmt.Sprintf(`<fusion-max-width width="%d"></fusion-max-width>`, i.size))
}

File: components/MaxWidth/ts/Component.ts
import {html, LitElement, unsafeCSS} from 'lit';
import {customElement, property} from 'lit/decorators.js';
// @ts-ignore
import styles from '../scss/Styles.scss';

@customElement('fusion-max-width')
export class FusionIcon extends LitElement {
    static styles = unsafeCSS(styles);

    @property({reflect: true})
    width = 1024

    attributeChangedCallback(name: string, _old: string | null, value: string | null) {
        super.attributeChangedCallback(name, _old, value);
        if (Number(value) < 1) {
            this.width = Number(_old)
        }
        this.style.setProperty('--max-width', this.width + "px")
    }

    render() {
        return html`
          <div class="container">
            <slot></slot>
          </div>`;
    }
}
